tools/cabana/streams/routes.cc

```diff
@@ -32,10 +32,25 @@ class RouteListWidget : public QListWidget {
 RoutesDialog::RoutesDialog(QWidget *parent) : QDialog(parent) {
   setWindowTitle(tr("Remote routes"));
 
+  auto all_routes_widget = new QWidget;
+  auto all_routes_layout = new QVBoxLayout;
+  all_routes_layout->addWidget(period_selector_ = new QComboBox(this));
+  all_routes_layout->addWidget(route_list_ = new RouteListWidget(all_routes_widget));
+  all_routes_widget->setLayout(all_routes_layout);
+
+  auto preserved_routes_widget = new QWidget;
+  auto preserved_layout = new QVBoxLayout;
+  preserved_layout->addWidget(preserved_route_list_ = new RouteListWidget(all_routes_widget));
+  preserved_routes_widget->setLayout(preserved_layout);
+
+  routes_type_selector_ = new QTabWidget(this);
+  routes_type_selector_->addTab(all_routes_widget, tr("&All"));
+  routes_type_selector_->addTab(preserved_routes_widget, tr("&Preserved"));
```

Comment:
You don't need to create an extra `preserved_routes_widget` and a layout just to add `preserved_route_list_` to the tab, since `preserved_route_list_` is already a widget. You can directly add it to the tab with a single line:

` routes_type_selector_->addTab(preserved_route_list_ = new RouteListWidget, tr("&Preserved"));`